proprietary/tokens/src/components/ams/icon.tokens.json

```diff
@@ -0,0 +1,24 @@
+{
+  "ams": {
+    "icon": {
+      "size": {
+        "3": {
+          "font-size": { "value": "{ams.text.level.3.font-size}" },
+          "line-height": { "value": "{ams.text.level.3.line-height}" }
+        },
+        "4": {
+          "font-size": { "value": "{ams.text.level.4.font-size}" },
+          "line-height": { "value": "{ams.text.level.4.line-height}" }
+        },
+        "5": {
+          "font-size": { "value": "{ams.text.level.5.font-size}" },
+          "line-height": { "value": "{ams.text.level.5.line-height}" }
+        },
+        "6": {
+          "font-size": { "value": "{ams.text.level.6.font-size}" },
+          "line-height": { "value": "{ams.text.level.6.line-height}" }
+        }
+      }
```

Comment:
I'm not sure this is an improvement. The icon variants are called `size-3`, `size-4`, etc [1]. Maybe you wanted to align it to the Heading props, but those variants are called `1`, `2`, etc. 

[1] The Icon `size` prop actually has `level-3`, `level-4`, etc. Maybe we should align those 2 as well? Maybe not in this PR though

Reply:
Whoah, did you actually read everything? 😮 

I think we can still split these names, but I agree that a separate task is better for it – even more so if we consider changing icon props anyway. Reverted.

Reply:
No, just checked the separate commits 😁 Sounds like a plan to me!